### DISCUSSION TITLE
Flet 打包APK 成功但是，引用requests 之后异常

### DISCUSSION THREAD

**Discussion by ChinaGodzilla** [flet-dev/flet#2835] [Q&A]
I wrote a routine that did not include the requests package and packaged the APK so that it could run normally on Android. When I referenced the requests library and packaged the APK, an error was reported as shown below.

![96930525-11ea-4021-964d-48879f1d3f98](https://github.com/flet-dev/flet/assets/43966525/0c6abfc7-89b2-40eb-88c9-e168f83396e8)

python code：

`import time

import flet as ft
import requests
from bs4 import BeautifulSoup
Url_list = []

def Get_news():
    Count = 0
    Web_text = requests.get('https://news.china.com/domestic/?x4fc.html')
    #print(Web_text.text)
    soup = BeautifulSoup(Web_text.content.decode('utf8'), 'html.parser')
    # 查找所有的<a>标签
    news_links = soup.find_all('a')

    # 遍历每个<a>标签，输出链接和文本内容
    for link in news_links:
        Count += 1
        href_value = link.get('href')
        news_text = link.text
        #print(f"Link: {href_value}, Text: {news_text}")
        if Count > 40:
            Url_list.append(news_text)

def main(page: ft.Page):
    page.title = "疼殉新闻"
    page.vertical_alignment = ft.MainAxisAlignment.CENTER
    page.appbar = ft.AppBar(title=ft.Text('疼殉新闻'),center_title=True,color='white',bgcolor=ft.colors.BLUE)
    page.navigation_bar = ft.NavigationBar(
        destinations=[
            ft.NavigationDestination(icon=ft.icons.HOME, label="首页"),
            ft.NavigationDestination(icon=ft.icons.NAVIGATION, label="热点"),
            ft.NavigationDestination(
                icon=ft.icons.ACCOUNT_BOX,
                selected_icon=ft.icons.BOOKMARK,
                label="我的",
            ),
        ]
    )

    lv = ft.ListView(expand=True, spacing=10)
    Get_news()

    for i in Url_list:
        lv.controls.append(ft.Text(str(i)))

    page.update()
    page.add(lv

    )

ft.app(main)`

Packaging command

flet build apk --verbose

**Comment by burhansvural** [comment]
There are errors and deficiencies in the codes. Additionally, with the "flet build apk" command, the .apk file was created for me without any errors.

Edited codes:
```python
import flet as ft
import requests
from bs4 import BeautifulSoup
Url_list = []

def Get_news():
    global Url_list
    Count = 0
    Web_text = requests.get('https://news.china.com/domestic/?x4fc.html')
    #print(Web_text.text)
    soup = BeautifulSoup(Web_text.content.decode('utf8'), 'html.parser')
    # 查找所有的标签
    news_links = soup.find_all('a')

    # 遍历每个<a>标签，输出链接和文本内容
    for link in news_links:
        Count += 1
        href_value = link.get('href')
        news_text = link.text
        #print(f"Link: {href_value}, Text: {news_text}")
        if Count > 40:
            Url_list.append(news_text)

def main(page: ft.Page):
    global Url_list
    page.title = "疼殉新闻"
    page.vertical_alignment = ft.MainAxisAlignment.CENTER
    page.appbar = ft.AppBar(title=ft.Text('疼殉新闻'),center_title=True,color='white',bgcolor=ft.colors.BLUE)
    page.navigation_bar = ft.NavigationBar(
    destinations=[
        ft.NavigationDestination(icon=ft.icons.HOME, label="首页"),
        ft.NavigationDestination(icon=ft.icons.NAVIGATION, label="热点"),
        ft.NavigationDestination(
            icon=ft.icons.ACCOUNT_BOX,
            selected_icon=ft.icons.BOOKMARK,
            label="我的",
        ),
    ])

    lv = ft.ListView(expand=True, spacing=10)
    page.add(lv)
    Get_news()
    for i in Url_list:
        lv.controls.append(ft.Text(str(i)))

    page.update()
    
ft.app(main)
```

**Comment by ChinaGodzilla** [comment]
@burhansvural thanks

**Reply by ChinaGodzilla** [reply]
@burhansvural 

Hello, I used the code and command line you provided for packaging, and the result is still as follows ：

![image](https://github.com/flet-dev/flet/assets/43966525/633363f5-cf4e-4b06-b4b5-367fbdc350d7)

run of phone ：

![image](https://github.com/flet-dev/flet/assets/43966525/15140e18-9740-4c89-9fa4-4fec48289b3b)


This is my build directory including source code：
[Buildapk.zip](https://github.com/flet-dev/flet/files/14546068/Buildapk.zip)


This is the APK file I built successfully：
https://www.sctulong.com/news.apk

**Comment by ChinaGodzilla** [comment]
@burhansvural  

The final check was that the requirements.txt file did not fill in the dependencies, thank you.

**Reply by burhansvural** [reply]
requirements.txt contents:
beautifulsoup4==4.12.3
flet==0.21.1
Requests==2.31.0

![newsapp](https://github.com/flet-dev/flet/assets/149883930/b20949e9-475a-4f35-8dcc-2a863d146458)

The pleasure was mine. I wish you success in your work.
I recommend you use the "pipreqs" command to automatically create requirements.txt...
